Tests/MongoSwiftTests/AsyncTestUtils.swift

```diff
@@ -73,9 +73,10 @@ extension MongoSwiftTestCase {
     internal func withTestClient<T>(
         _ uri: String = MongoSwiftTestCase.getConnectionString().toString(),
         options: MongoClientOptions? = nil,
+        eventLoopGroup: MultiThreadedEventLoopGroup? = nil,
```

Comment:
It accepts a `MultiThreadedEventLoopGroup` since this method already uses one. I am not sure why `makeTestClient` just accepts an `EventLoopGroup` though. I wonder if we want to do that here too.

Reply:
`EventLoopGroup` is the protocol that `MultiThreadedEventLoopGroup` conforms to. I think we use it in `makeTestClient` in case we want to try mock an event loop or something. I think we ought to just use the protocol here too, though I don't anticipate using a non-`MutliThreadedEventLoopGroup` for a while.